File: Java/Easy/MergeStringsAlternately.java
class MergeStringsAlternately {

    public static void main(String[] args) {
        String result = mergeAlternately("abc", "pqr");

        System.out.println(result);

    }

    public static String mergeAlternately(String word1, String word2) {
        String result = "";
        int len = word1.length() > word2.length() ? word1.length() : word2.length();

        for (int i = 0; i < len; i++) {
            if (!(i >= word1.length()))
                result += (word1.charAt(i));
            if (!(i >= word2.length()))
                result += (word2.charAt(i));
        }

        return result;

    }
}

File: Python/Easy/MergeStringsAlternately.py

'''
Input: word1 = "abc", word2 = "pqr"
Output: "apbqcr"
Explanation: The merged string will be merged as so:
word1:  a   b   c
word2:    p   q   r
merged: a p b q c r

'''

def mergeAlternately(word1, word2): 
    """
    :type word1: str
    :type word2: str
    :rtype: str
    """
    result = ""
    length = min(len(word1), len(word2))

    for i in range(length):
        result += word1[i] + word2[i]

    if len(word1) > len(word2):
        result += word1[length:]
        return result
    result += word2[length:]
    return result

print(mergeAlternately("abc", "pqr"))

# Comment